pepper-apis/src/main/java/org/broadinstitute/ddp/service/Auth0LogEventService.java

```diff
@@ -0,0 +1,112 @@
+import static org.broadinstitute.ddp.json.auth0.Auth0LogEventNode.resolveDateTimeValue;
+import static org.broadinstitute.ddp.json.auth0.Auth0LogEventNode.resolveStringValue;
+import static org.slf4j.LoggerFactory.getLogger;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import org.broadinstitute.ddp.db.TransactionWrapper;
+import org.broadinstitute.ddp.db.dao.Auth0LogEventDao;
+import org.broadinstitute.ddp.util.GsonRecursiveReader;
+import org.slf4j.Logger;
+
+/**
+ * The service used for handling Auth0 log events which fired by Auth0 streams
+ * (see https://manage.auth0.com/dashboard/us/ddp-dev/log-streams).<br>
+ * <b>The service provides the following functions:</b>
+ * <ul>
+ *     <li>parses JSON doc (coming as a payload in POST request specified in Auth0 Custom Webhook)
+ *     extracting log events data from it;</li>
+ *     <li>saves extracted log events to DB table 'auth0_log_event';</li>
+ *     <li>adds extracted log events to log (together with tenant name detected from POST URL).</li>
+ * </ul>
+ * NOTE: see Auth0LogEventServiceTest for examples of Auth0 log events JSON docs.
+ */
+public class Auth0LogEventService {
+
+    private final Logger log = getLogger(this.getClass());
+
+    /**
+     * List of Auth0 Log Event JSON node names which to read (recursively)
+     */
+    static final List<String> AUTH0_LOG_EVENT_NODE_NAMES = List.of(
+            LOG_ID.nodeName(),
+            DATE.nodeName(),
+            TYPE.nodeName(),
+            CLIENT_ID.nodeName(),
+            CONNECTION_ID.nodeName(),
+            USER_ID.nodeName(),
+            USER_AGENT.nodeName(),
+            IP.nodeName(),
+            EMAIL.nodeName(),
+            USER_NAME.nodeName(),
+            DATA.nodeName()
+    );
+
+    /**
+     * Parses JSON doc (coming as a payload in POST request specified in Auth0 Custom Webhook)
+     * extracting log events data from it. Data fetched according to names
+     * defined in {@link #AUTH0_LOG_EVENT_NODE_NAMES}.
+     *
+     * @param logEventsJson payload doc coming with POST request (from Auth0 stream).
+     * @return {@code List<Map<String, JsonElement>>} list of maps with parsed log events data
+     */
+    public List<Map<String, JsonElement>> parseAuth0LogEvents(String logEventsJson) {
+        var rootNode = new Gson().fromJson(logEventsJson, JsonObject.class);
+        var logsNode = rootNode.getAsJsonArray(LOGS.nodeName());
+        final List<Map<String, JsonElement>> logEvents = new ArrayList<>();
+        for (JsonElement node : logsNode) {
+            logEvents.add(GsonRecursiveReader.readValues(node, AUTH0_LOG_EVENT_NODE_NAMES));
+        }
+        return logEvents;
+    }
+
+    public void logAuth0LogEvent(final Map<String, JsonElement> logEvent, String tenant) {
+        log.info(
```

Comment:
I'm concerned that logging these entries will produce too much logging output and clog up stackdriver with noise. Might be better to change this to a "debug" log or something.

Reply:
Agreed, let's set this to `debug`.

Reply:
Changed to debug